linker/src/lib.rs

```diff
@@ -185,7 +187,9 @@ fn process_link(link: Link) -> PilStatement {
     } else {
         // plookup lhs is `flag { operation_id, inputs, outputs }`
         let lhs = SelectedExpressions {
-            selector: Some(from.flag),
+            selector: from
+                .instr_flag
+                .map_or(Some(from.link_flag.clone()), |f| Some(f * from.link_flag)),
```

Comment:
This is the same code as above (and I think I have seen similar code somewhere else) - can this be a function?